### ISSUE TITLE
chef_gem compile_time installation is deprecated

### ISSUE THREAD

**Issue by david-crowder** [cerner/cerner_splunk#49]
The [12.1.0](https://www.chef.io/blog/2015/03/03/chef-12-1-0-released/) release logs a warning for compile time chef_gem installations. The release blog states that the behavior will change in 13 so that compile_time defaults to false and that chef_gem resources should explicitly set the compile_time attribute appropriately (or, more favorably, be refactored).
``` 
chef_gem[chef-vault] chef_gem compile_time installation is deprecated
```

**Comment by acharlieh** [comment]
The issue at hand is that we use chef-vault at compile time to read encrypted configuration (and validate the same, and drive creation of resources with encrypted data.) I think we should start with the explicit (backwards compatible) setting of the attribute for now, and then see what other options exist for handling encrypted data in the future.

**Comment by acharlieh** [comment]
We should also look at the license_server recipe and its use of nokogiri. That one we may be able to refactor into a resource to manage splunk licenses.

**Comment by acharlieh** [comment]
Marking as wontfix, With the change merged in e172c1d999793963d4e51d8bdfb38e36402f8eea chef_gem is no longer called directly by us.
